### ISSUE TITLE
Ideally this dataset should be made more "modular" and lightweight

### ISSUE THREAD

**Issue by yarikoptic** [ReproBrainChart/HBN_CPAC#2]
Initially brought up attention to this by https://neurostars.org/t/cloning-hbn-cpac-data/30587/4 .

I cloned this dataset locally to investigate and found that it consumes 1.3G of `.git/objects` and has over 4M files in the tree. That renders this dataset **very hard to use**. E.g. I am still (after an hour?) waiting for `git annex get`  on a single sample file after the initial `git clone`.

Ideally individual datalad datasets (git/git-annex) repos should have up to few hundred thousand files in a tree (not millions), and then use `git submodule` mechanism to compose into a larger dataset. That allows for easy reuse of individual components etc. An example of such dataset could be https://github.com/dandisets/000108 which covers over 332M files throughout its individual subdatasets, which are some of the https://github.com/dandizarrs/ ones.  More on this could be read in https://handbook.datalad.org/en/latest/beyond_basics/101-160-gobig.html  and there about.

I think it would make a total sense to make each subject into a subdataset. That would indeed make it into a significant number of subdatasets (over 2k) with each about 2k files:

```shell
$> git ls-tree -r HEAD >| /tmp/hbn-tree.txt
$> sed -ne '/cpac_R/s,.*cpac_RBCv0/,,g' -e 's,/ses-.*,,gp' /tmp/hbn-tree.txt| sort | uniq -c | nl | tail
  2447     2336 sub-NDARZW262ZLV
  2448     2008 sub-NDARZW363UGM
  2449     2336 sub-NDARZW472CCF
  2450     2008 sub-NDARZW619BDL
  2451     2336 sub-NDARZW873DN3
  2452     2008 sub-NDARZX163EWC
  2453     2336 sub-NDARZY101JNB
  2454     2336 sub-NDARZY668NMV
  2455     2008 sub-NDARZZ740MLM
  2456     1680 sub-NDARZZ810LVF
```

but overall it would make such dataset much more usable IMHO.

**Comment by mattcieslak** [comment]
We had a hard time figuring out how to keep all the git history and run this with FairlyBig. We were also not sure if github would us allow to make thousands of repos for the subdatasets (It looks like you were able to do this with dandi zarrs, though). 

Is there a way to split these into subdatasets and keep the history?

We've tested this current inefficient configuration and it definitely is slow. But, downloading more optimized datasets like HCP-openaccess also takes a very long time when you download all the subdatasets and their contents. The RBC CPAC datasets are only very slow if you want to interactively get a single subject or handful of files

**Comment by yarikoptic** [comment]
That's true -- likely if you would like to get an entire thing, a monolythic one would be "as good" or even more efficient. 

> Is there a way to split these into subdatasets and keep the history?

we now have all needed functionality implemented in git-annex (and git) but do need to provide `datalad` level convenience functionality. I will add it to my plate to see implemented "now" (i.e. within coming weeks likely after sfn) . Please review a potential interface:

- https://github.com/datalad/datalad/issues/3554#issuecomment-1202591654
